source/frontend/src/features/mapSideBar/compensation/list/CompensationListContainer.tsx

```diff
@@ -121,26 +145,56 @@ export const CompensationListContainer: React.FC<ICompensationListContainerProps
       legacyPayee: null,
       isPaymentInTrust: null,
       gstNumber: null,
-      compensationRequisitionProperties:
-        fileProperties?.map(x => {
-          return {
-            compensationRequisitionPropertyId: null,
-            compensationRequisitionId: null,
-            propertyAcquisitionFileId: x.id,
-            acquisitionFileProperty: null,
-          } as ApiGen_Concepts_CompensationRequisitionProperty;
-        }) || [],
+      compReqLeaseStakeholder: null,
+      compReqAcquisitionProperties: null,
+      compReqLeaseProperties: null,
       ...getEmptyBaseAudit(),
     };
 
-    postAcquisitionCompensationRequisition(fileId, defaultCompensationRequisition).then(
-      async newCompensationReq => {
-        if (newCompensationReq?.id) {
-          sidebar.setStaleLastUpdatedBy(true);
-          sidebar.setStaleFile(true);
-        }
-      },
-    );
+    switch (fileType) {
+      case ApiGen_CodeTypes_FileTypes.Acquisition:
+        newCompensationRequisition.acquisitionFileId = fileId;
+        newCompensationRequisition.leaseId = null;
+        newCompensationRequisition.compReqAcquisitionProperties =
+          file.fileProperties?.map(x => {
+            return {
+              compensationRequisitionPropertyId: null,
+              compensationRequisitionId: null,
+              propertyAcquisitionFileId: x.id,
+              acquisitionFileProperty: null,
+            } as ApiGen_Concepts_CompReqAcquisitionProperty;
+          }) || [];
+        break;
+      case ApiGen_CodeTypes_FileTypes.Lease:
+        newCompensationRequisition.leaseId = fileId;
+        newCompensationRequisition.acquisitionFileId = null;
+        newCompensationRequisition.compReqLeaseProperties =
+          file.fileProperties?.map(x => {
+            return {
+              compensationRequisitionPropertyId: null,
+              compensationRequisitionId: null,
+              propertyLeaseId: x.id,
+              leaseProperty: null,
+            } as ApiGen_Concepts_CompReqLeaseProperty;
+          }) || [];
+        break;
+      default:
+        return null;
+    }
+
+    return newCompensationRequisition;
+  };
+
+  const onAddCompensationRequisition = () => {
+    createCompensationRequisition(
+      fileType,
+      getDefaultCompensationRequisition(fileType, file.id),
+    ).then(async newCompensationReq => {
+      if (newCompensationReq?.id) {
```

Comment:
what does this look like in the case of an error?

Reply:
Added try/catch